grpcv1.go

```diff
@@ -106,11 +106,24 @@ func (api headscaleV1APIServer) CreatePreAuthKey(
 		expiration = request.GetExpiration().AsTime()
 	}
 
+	if len(request.AclTags) > 0 {
+		for _, tag := range request.AclTags {
+			err := validateTag(tag)
+
+			if err != nil {
+				return &v1.CreatePreAuthKeyResponse{
+					PreAuthKey: nil,
+				}, status.Error(codes.InvalidArgument, err.Error())
+			}
+		}
+	}
```

Comment:
```suggestion
	for _, tag := range request.AclTags {
		err := validateTag(tag)

		if err != nil {
			return &v1.CreatePreAuthKeyResponse{
				PreAuthKey: nil,
			}, status.Error(codes.InvalidArgument, err.Error())
		}
	}
```
We don't need to check for empty lists as `range` will just not have anything to iterate over.